src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -520,14 +520,14 @@ export class TerminalService extends Disposable implements ITerminalService {
 	}
 
 	private _attachProcessLayoutListeners(): void {
```

Comment:
This looks like a method that's called repeatedly for the lifecycle of this service. If that's true `_register` must not be used, instead a session disposable that clears whatever a previous invocation has store should be used

Reply:
This is actually only called once here:

https://github.com/microsoft/vscode/blob/f933441a9e17555bb06fa98a29c199e559b4bf9b/src/vs/workbench/contrib/terminal/browser/terminalService.ts#L309-L313

The reconnect happens once per session and only for the "primary" backend, ie. you cannot reconnect to local terminals in a remote workspace.

Reply:
You know best, but `initializePrimaryBackend` is a public method and callable by anyone

Reply:
Made it private 👍